py/tests/test_solver.py

```diff
@@ -46,10 +46,12 @@ def test_managing_edit_variable() -> None:
     assert not s.hasEditVariable(v1)
     s.addEditVariable(v1, "weak")
     assert s.hasEditVariable(v1)
-    with pytest.raises(DuplicateEditVariable):
+    with pytest.raises(DuplicateEditVariable) as e:
         s.addEditVariable(v1, "medium")
-    with pytest.raises(UnknownEditVariable):
+    assert e.value.edit_variable is v1
```

Comment:
Is accessing `.value` the right thing here? Shouldn't it just be `e.edit_variable`?

Reply:
``pytest.raises`` return a custom object ``ExcInfo``. ``value`` is the way to reach the original exception.